examples/generative/speaker_recognition_using_cnn.py

```diff
@@ -0,0 +1,465 @@
+are longer than 1 second (and originally not sampled at 16000 Hz, but we assume that by at
+this stage, you would have resampled them to 16000 Hz). We will use those 6 files to create
+354 1-second-long noise samples to be used for training
+
+Let's sort these 2 categories into 2 folders:
+
+- An `audio` folder which will contain all the per-speaker speech sample folders
+- A `noise` folder which will contain all the noise samples
+"""
+
+"""
+Before sorting the audio and noise categories into 2 folders,
+we have the following directory structure:
+    main_directory/
+    ...speaker_a/
+    ...speaker_b/
+    ...speaker_c/
+    ...speaker_d/
+    ...speaker_e/
+    ...other/
+    ..._background_noise_/
+After sorting, we end up with the following structure:
+    main_directory/
+    ...audio/
+    ......speaker_a/
+    ......speaker_b/
+    ......speaker_c/
+    ......speaker_d/
+    ......speaker_e/
+    ...noise/
+    ......other/
+    ......_background_noise_/
+"""
+
+# If folder `audio`, does not exist, create it, otherwise do nothing
+if os.path.exists(DATASET_AUDIO_PATH) is False:
+    os.makedirs(DATASET_AUDIO_PATH)
+
+# If folder `noise`, does not exist, create it, otherwise do nothing
+if os.path.exists(DATASET_NOISE_PATH) is False:
+    os.makedirs(DATASET_NOISE_PATH)
+
+for folder in os.listdir(DATASET_ROOT):
+    if os.path.isdir(os.path.join(DATASET_ROOT, folder)):
+        if folder in [AUDIO_SUBFOLDER, NOISE_SUBFOLDER]:
+            # If folder is `audio` or `noise`, do nothing
+            continue
+        elif folder in ["other", "_background_noise_"]:
+            # If folder is one of the folders that contains noise samples,
+            # move it to the `noise` folder
+            shutil.move(
+                os.path.join(DATASET_ROOT, folder),
+                os.path.join(DATASET_NOISE_PATH, folder),
+            )
+        else:
+            # Otherwise, it should be a speaker folder, then move it to
+            # `audio` folder
+            shutil.move(
+                os.path.join(DATASET_ROOT, folder),
+                os.path.join(DATASET_AUDIO_PATH, folder),
+            )
+
+"""
+## Noise preparation
+
+In this section:
+
+- We load all noise samples (which should have been resampled to 16000)
+- We split those noise samples to chuncks of 16000 samples which
+correspond to 1 second duration each
+"""
+
+# Get the list of all noise files
+noise_paths = []
+for subdir in os.listdir(DATASET_NOISE_PATH):
+    subdir_path = Path(DATASET_NOISE_PATH) / subdir
+    noise_paths += [
+        os.path.join(subdir_path, filepath)
+        for filepath in os.listdir(subdir_path)
+        if filepath.endswith(".wav")
+    ]
+
+print(
+    "Found {} files belonging to {} directories".format(
+        len(noise_paths), len(os.listdir(DATASET_NOISE_PATH))
+    )
+)
+
+"""
+Resample all noise samples to 16000 Hz
+"""
+
+"""shell
+!dataset_noise_path=~/Downloads/16000_pcm_speeches/noise; \
```

Comment:
Don't prefix lines with `!`, the `!` is added automatically when running the example

Reply:
Thank you for the review and approval.
One quick question before I upload the remaining files:
Will the script add ! To each line? In this case I think it won’t work cause it is a series of shell commands that I have split into lines using \ in order for the line not to be extremely huge

Reply:
Yeah, that's unfortunate. I would suggest using `os.system` to run the command line in a Python cell in this case.

Reply:
Thanks! Will do this.
Shall I still upload all files? Or only .py?

Reply:
Sure, please add the generated files.